nixos/modules/services/web-servers/stargazer.nix

```
@@ -0,0 +1,197 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.stargazer;
+  routesFormat = pkgs.formats.ini { };
+  globalFile = pkgs.writeText "global.ini" ''
+    listen = ${concatStringsSep " " cfg.listen}
+    connection-logging = ${boolToString cfg.connectionLogging}
+    log-ip = ${boolToString cfg.ipLog}
+    log-ip-partial = ${boolToString cfg.ipLogPartial}
+    request-timeout = ${toString cfg.requestTimeout}
+    response-timeout = ${toString cfg.responseTimeout}
+
+    [:tls]
+    store = ${toString cfg.store}
+    organization = ${cfg.certOrg}
+    gen-certs = ${boolToString cfg.genCerts}
+    regen-certs = ${boolToString cfg.regenCerts}
+    ${if cfg.certLifetime == "0" then "" else cfg.certLifetime}
```

Shouldn't this be a number?

---

No, the value is supposed to be an amount of time like 3y for 3 years. There is always a letter at the end. The only way to have an infinate lifetime is to not include that parameter at all. Is there a better way to do an optional parameter?

---

Oh, the time parameter is written as a string? Curious.

Nonetheless, I believe you should use optionalString:

 ```suggestion
    ${optionalString (cfg.certLifetime != "0") cfg.certLifetime}
```

P.S.: shouldn't this be something like 

 ```suggestion
    cert-lifetime = ${optionalString (cfg.certLifetime != "0") cfg.certLifetime}
```

?

---

"0" is just a placeholder value that tells it not to put that option in the config file.

---

Ah yes, my mistake.